frontend/packages/console-dynamic-plugin-sdk/README.md

```diff
@@ -266,6 +268,32 @@ It provides access to specific modules exposed by the plugin. It's loaded right
 `exposed-barUtils-chunk.js` is the generated webpack chunk for `barUtils` exposed module. It's loaded
 via the plugin entry chunk (`plugin-entry.js`) when needed.
 
+Plugins may also include other assets, such as JSON localization files that follow the general pattern
+`locales/<lang>/plugin__<plugin-name>.json` or static images referenced from the plugin code.
+
+## Serving plugin assets
+
+Dynamic plugins are deployed on a cluster via OLM operators. Each plugin deployment should include a web
+server that hosts the [generated assets](#generated-assets) of the given plugin.
+
+**Important!** Make sure to configure your plugin web server to use MIME type `text/javascript` for all
+JS assets to avoid any issues with the `X-Content-Type-Options: nosniff` security header used by Console
+when fetching plugin assets via Bridge URL `/api/plugins/<plugin-name>`.
```

Comment:
Agreed, will update.

Reply:
just confirming: is it alright if plugin's [web server](https://github.com/red-hat-storage/odf-operator/blob/main/console/nginx_conf.go#L54-L83) adds `application/javascript` by default instead of explicitly `text/javascript` ?? (so far it is working)...

Reply:
@SanjalKatiyar Thanks, this actually led me to realize what the core issue was.

Check out https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options

Console [loads](https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/runtime/plugin-loader.ts) dynamic plugin scripts by injecting `<script>` elements into the main DOM document.

Console Bridge server [adds](https://github.com/openshift/console/blob/master/pkg/server/middleware.go) `X-Content-Type-Options: nosniff` HTTP response header to all plugin asset requests.

Web browsers that comply with `X-Content-Type-Options: nosniff` header will block `<script>` initiated requests when the MIME type of requested asset is not [valid](https://mimesniff.spec.whatwg.org/#javascript-mime-type).

`application/javascript` is considered to be a [legacy JavaScript MIME type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#legacy_javascript_mime_types) but is still a valid one.

Looking further into this, I think it's OK to use a legacy MIME type like `application/javascript`.

**Note that the proper MIME type for JavaScript according to current HTML spec is `text/javascript`.**

The core of the issue is "always make sure to provide a valid mime type for JS assets" so that you won't end up facing errors in the browser like this one
```
Refused to execute script from 'http://localhost:9000/api/plugins/foo-plugin/some-script.js' because its MIME type ('') is not executable, and strict MIME type checking is enabled.
```

I will update the README accordingly.

Reply:
thanks !